OpenRA.Game/WorldUtils.cs

```diff
@@ -41,6 +41,113 @@ public static IEnumerable<Actor> FindActorsInCircle(this World world, WPos origi
 			}
 		}
 
+		/// <summary>
+		/// Finds all the actors of which their health radius is intersected by a ray (with a definable width) between two points.
+		/// TODO: We should overselect the area, since actors that ight intersect may be larger than 1 cell.
+		/// TODO: Update this to take the z offsets into account, allowing for 3D raycasting.
+		/// </summary>
+		/// <param name="world">The engine world the raycast is to be done in.</param>
+		/// <param name="target">The position the ray should end at</param>
+		/// <param name="source">The position the ray should start at</param>
+		/// <param name="rayWidth">How close an actor's health radius needs to be to the ray to be considered 'intersected' by the ray</param>
+		/// <returns>A list of all the actors 'intersected' by the ray</returns>
+		public static IEnumerable<Actor> FindActorsInRaycast(this World world, WPos target, WPos source, WDist rayWidth)
+		{
+			// This raycast is done by first just finding all actors within a square that starts at the source, and ends at the target.
+			// Then we iterate over this list, and find all actors for which their health radius is at least within rayWidth of the ray.
+			// For actors without a health radius, we simply check their center point.
+			// The square in which we select all actors must be large enough to encompass the entire ray's width.
+			var xOver = 0;
```

Comment:
`var xMargin = Math.Sign(target.X - source.X) * 1024;` and likewise for y. (over &rarr; margin makes the purpose much clearer)